### PR TITLE
Misc test updates

### PR DESCRIPTION
**Pull request by rejeep**
This adds a testing sandbox to the unit testing environment so that's easier to create tests with real directories/files.

### CODE REVIEW

File: mocha.el [hunk 1]
```diff
@@ -17,6 +17,7 @@
 ;;; Code:
 (require 'compile)
 (require 'js2-mode)
+(require 'f)
```
Comment (scottaj) [inline mocha.el]: It looks like we didn't end up using this in the source file?

Comment (rejeep) [inline mocha.el]: Ahh, this is just the foundation to the other two PRs (that do need this). Do you want me to rewrite the history or should we let this slip through?

Comment (scottaj) [inline mocha.el]: Nope, I just looked at this one first. Since we are using it in the other two anyways, I am fine with leaving it.

**Comment by scottaj** [comment]
This LGTM, but I think I just created merge conflicts on all three of your PRs by merging the earlier PR.

**Comment by rejeep** [comment]
Right, I will rebase them!

**Pull request merged**